src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -953,15 +956,35 @@ public Optional<File> findRapidScanConfig() throws OperationException {
                 .findRapidScanConfig(directoryManager.getSourceDirectory())
         );
     }
-    
+
+    private int countBdioEntryFiles() throws IntegrationException {
+        File bdioFile = this.fileFinder.findFile(directoryManager.getBdioOutputDirectory(), "*.bdio");
+        Bdio2ContentExtractor bdio2Extractor = new Bdio2ContentExtractor();
+        return bdio2Extractor.extractContent(bdioFile).size() - 1; // excludes bdio header file
+    }
+
+    public int calculateMaxWaitInSeconds(int bdioEntriesCount) {
+        // Max polling interval time will be the (N+1)th Fibonacci number in seconds, where N is the number of BDIO chunks (entry files)
+        int maxWaitInSeconds = 1;
+        int[] customFibonacciSequence = {0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55};
```

Comment:
These should be class level final fields.
maxWaitInSeconds should get its default value from a configurable optional property called --min-wait-in-seconds, which in turn should have the default value of 1.

Reply:
Oh was a new Detect property also a part of this current requirement? I was under the impression that the "configurable" part was with respect to the weight (N+w) logic from our [design doc](https://sig-confluence.internal.synopsys.com/display/integration/Configurable+Polling+Interval+in+Communication) and that the configurable weight part is for the future. Design doc and JIRA both currently state introducing a property as a future change. Lmk!